### PR TITLE
Update schema.json and implement minor compliance change in apps.json

### PR DESCRIPTION
**Pull request by kingthorin**
Add GitHub action to run on every PR that changes apps.json, causing apps.json to be validated against schema.json.

To help prevent things like:
https://github.com/AliasIO/wappalyzer/pull/3008

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

File: schema.json [hunk 1]
```diff
@@ -2,18 +2,18 @@
 	"title": "Wappalyzer Schema",
 	"type": "object",
 	"additionalProperties": false,
+	"required": ["categories","apps"],
```
Comment (kingthorin) [inline schema.json:5]: Use JSON schema 4+ required directive. `categories` and `apps` are required.

File: schema.json [hunk 2]
```diff
@@ -2,18 +2,18 @@
...
 	"properties": {
 		"$schema": {
 			"type": "string"
 		},
 		"categories": {
 			"type": "object",
-			"required": true,
+			"minProperties": 64,
```
Comment (kingthorin) [inline schema.json:12]: New categories aren't added often. IDs `40` and `43` are missing, therefore there are currently 64 expected.

File: schema.json [hunk 3]
```diff
@@ -27,89 +27,136 @@
 		},
 		"apps": {
 			"type": "object",
-			"required": true,
 			"additionalProperties": {
 				"additionalProperties": false,
+				"required": ["cats","website"],
```
Comment (kingthorin) [inline schema.json:32]: Each `app` requires `cats` and `website`

File: schema.json [hunk 4]
```diff
@@ -27,89 +27,136 @@
...
 				"properties": {
 					"cats": {
 						"type": "array",
 						"items": {
 							"type": "number"
 						},
-						"required": true
+						"minItems": 1
```
Comment (kingthorin) [inline schema.json:39]: At least one category

File: schema.json [hunk 5]
```diff
@@ -27,89 +27,136 @@
...
 					},
 					"cpe": {
-						"type": "string"
+						"type": "string",
+						"pattern": "^(?!\\s*$).+"
```
Comment (kingthorin) [inline schema.json:43]: Non-blank/empty string.

File: schema.json [hunk 6]
```diff
@@ -27,89 +27,136 @@
...
 					},
 					"cookies": {
 						"type": "object",
-						"additionalProperties": {
-							"type": "string"
+						"additionalProperties": false,
+						"patternProperties": {
+							"^.+$": {
+							},
```
Comment (kingthorin) [inline schema.json:50]: Cookies have to have a name?

File: schema.json [hunk 7]
```diff
@@ -27,89 +27,136 @@
...
 					"cpe": {
-						"type": "string"
+						"type": "string",
+						"pattern": "^(?!\\s*$).+"
 					},
 					"cookies": {
 						"type": "object",
-						"additionalProperties": {
-							"type": "string"
+						"additionalProperties": false,
```
Comment (kingthorin) [inline schema.json:47]: Per following lines: name and optional value, nothing else.

File: schema.json [hunk 8]
```diff
@@ -27,89 +27,136 @@
...
+								"items": {
+									"type": "string",
+									"pattern": "^(?!\\s*$).+"
+								}
+							},
+							{
+								"type": "string",
+								"pattern": "^(?!\\s*$).+"
+							}
+						]
```
Comment (kingthorin) [inline schema.json:91]: Can be an arrary of non-blank/empty strings or simply a non-blank/empty string.

**Comment by kingthorin** [comment]
Added comments explaining the changes.

I had done the test setup in this repo: https://github.com/kingthorin/fred/actions

I'll leave it alive until after this is merged in case you want to look at the behavior.

You can also use: https://www.jsonschemavalidator.net/ to do some testing/validation if you like.

**Comment by AliasIO** [comment]
We're already using Travis CI, which validates apps.json against the schema. I'd be happy to pull in your schema changes though.

**Comment by kingthorin** [comment]
It's missing things though, like that spurious "env" entry in Clickfunnels.

It's your call of course. I'm glad to remove the workflow if you want to use the schema.json and apps.json contribution.

**Comment by AliasIO** [comment]
I think the whole development, testing and deployment pipeline needs an overhaul. I just haven't had the time for it. For now I think making the schema validation work properly in Travis is the best option.

**Comment by kingthorin** [comment]
Fair enough, I'll remove the workflow. Update inbound 😀

**Comment by AliasIO** [comment]
Thanks!

**Pull request merged**